### DISCUSSION TITLE
How do I know if my plugin is loaded?

### DISCUSSION THREAD

**Discussion by kincsescsaba** [scala/scala3#21238] [Compiler internals Q&A] [answered]
Hi there,

I had a working plugin and could not isolate why it had stopped working. There are no error messages. What I see is that I can add the plugin with `-Xplugin:path/to/jar`, and there are no error messages, but the plugin does not do its job of filtering specific code parts that worked previously.

I tried `scala -Xplugin:path/to/jar -Xplugin-list` options, this returns an empty line, then quits.

What are the best practices to resolve an issue like this?

**Comment by som-snytt** [comment, marked as answer]
```
  -Xplugin-require:<plugin>        Abort if a named plugin is not loaded.
           -Xplugin-require  Abort if a named plugin is not loaded.
```
both compilers have the option. I think a plugin should issue diagnostics if it decides not to run.

**Reply by kincsescsaba** [reply]
It looks like my specific issue was caused by refactoring a package name. I did not notice that the IDE did not change the name in build.properties, hence the missing plugin.
You're right. `-Xplugin-require should display more about plugin loading errors, but I found the edge case where there is no error message. In my personal opinion, it would be nice if this issue failed with an explicit error.
